src/runtime/hexagon/hexagon_vtcm_pool.cc

```diff
@@ -85,7 +85,7 @@ void* HexagonVtcmPool::Allocate(size_t nbytes) {
 
   auto entry_to_allocate = free_.begin();
   for (auto it = free_.begin(); it != free_.end(); it++) {
-    if ((it->second < entry_to_allocate->second) && (it->second >= nbytes)) {
+    if (((entry_to_allocate->second < nbytes) || (it->second < entry_to_allocate->second)) && (it->second >= nbytes)) {
```

Comment:
The groups should be slightly adjusted:

`if ((entry_to_allocate->second < nbytes) || ((it->second < entry_to_allocate->second) && (it->second >= nbytes))) {`

Reply:
Doesnt this lead to us updating entry_to_allocate for iterator values that may not fit the desired allocation?

Reply:
But, that's ok because it will still find a free block that works if there is one.  But on reflection, I like your grouping better - I think it makes it more clear for future editors.